moped-editor/src/views/projects/projectView/ProjectComponents/index.js

```diff
@@ -52,9 +67,12 @@ export default function MapView({ projectName, projectStatuses }) {
   const appBarHeight = useAppBarHeight();
   const classes = useStyles({ appBarHeight });
   const mapRef = useRef();
+  const { projectId } = useParams();
 
   /* holds this project's components */
   const [components, setComponents] = useState([]);
+  const [featureCollectionsByComponentId, setFeatureCollectionsByComponentId] =
```

Comment:
I'm open to other ideas here, but this piece of state made it easier to get a component's feature collection and make good use of the `project_geography` view.

Another direction would be making feature collections on the fly from component records in the `components` state array, but that would also require calling `.find` each time. IMO it will take projects with tons of components to find out what works best.

Reply:
It's got to be kept somewhere, right? I don't feel like there's anything wrong with this, and given that the geometry is not a fluid, changing thing on the database, I totally want to get-it-once and hang onto it. 

Now, I do think that now that you've nailed down the shape of the feature collection that is most useful to your app, I want to take a swing at making the view generate that exact shape so one could just get the data out of the graphql endpoint and start using it without having to transmogrify it. Nothing urgent, as your system here is primo -- it would be a practice exercise in SQL more than anything.

Reply:
Agreed, y'all! I was hoping that the code would guide us to some thoughts on what shape we could get out of the DB down the road. 💪